ui/app/AppLayouts/Onboarding2/KeycardCreateProfileFlow.qml

```diff
@@ -175,15 +185,38 @@ SQUtils.QObject {
         id: keycardCreatePinPage
 
         KeycardCreatePinPage {
+            id: createPinPage
+
             onKeycardPinCreated: (pin) => {
+                createPinPage.loading = true
                 Backpressure.debounce(root, root.keycardPinInfoPageDelay, () => {
                     root.keycardPinCreated(pin)
-                    if (d.withNewSeedphrase)
-                        root.stackView.push(backupSeedIntroPage)
-                    else
-                        root.stackView.push(seedphrasePage)
                 })()
             }
+            Connections {
+                target: root
+                onKeycardStateChanged: function () {
+                    if (root.keycardState === Onboarding.KeycardState.NotEmpty) {
+                        if (d.withNewSeedphrase) {
+                            // Need to authorize before getting a seedphrase
+                            root.authorizationRequested()
```

Comment:
This doesn't look correct. If the authorization doesn't require any input from from the user from this page and is not caused by any action triggered there, that request shouldn't be generated be the UI. In other words, it should not be the responsibility of the UI, which should be unaware of this process.